### PR TITLE
ENH: Allow to run AttenuationImageFilter without a mask

### PR DESCRIPTION
**Pull request by dzenanz**
This computes attenuation estimates for all pixels in the image.

### CODE REVIEW

File: test/itkAttenuationImageFilterTest.cxx [hunk 1]
```diff
@@ -48,7 +48,12 @@ itkAttenuationImageFilterTest(int argc, char * argv[])
   using MaskImageType = itk::Image<LabelType, Dimension>;
 
   SpectraImageType::Pointer inputImage = itk::ReadImage<SpectraImageType>(std::string(argv[1]));
-  MaskImageType::Pointer    maskImage = itk::ReadImage<MaskImageType>(std::string(argv[2]));
+  MaskImageType::Pointer    maskImage;
+  std::string               maskFilename = argv[2];
+  if (maskFilename != "nul")
```
Comment (tbirdso) [inline test/itkAttenuationImageFilterTest.cxx:53]: Nitpick, "nul" or "null"?

**Review by tbirdso — APPROVED** [review]
Had to reread the `m_LastScanlineIndex` check but it makes sense now. LGTM.

File: test/itkAttenuationImageFilterTest.cxx (hunk 1 above)
Comment (dzenanz) [reply test/itkAttenuationImageFilterTest.cxx:53]: `NUL`, thinking of Windows special file. I should have maybe spelled it with uppercase letters.

Comment (tbirdso) [reply test/itkAttenuationImageFilterTest.cxx:53]: Got it, thanks for the explanation.

**Pull request merged**